main.cpp

```diff
@@ -250,7 +487,10 @@ struct ping_queue_t
 
 		*out = m_queue.front();
 		m_queue.pop_front();
-		m_ips.erase(out->ep);
+		if (out->ep.protocol() == udp::v4())
```

Comment:
It's only two extra if statements over the template solution. Templating doesn't seem worth the extra annoyances like having to break endpoints into separate address_type, port pairs and checking two queues after every packet.